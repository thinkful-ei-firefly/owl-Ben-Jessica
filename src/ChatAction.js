import React from 'react';

function ChatAction(props) {
  let action;
  let icon;
  const type = props.type;
  switch (type) {
    case 'thumbs-up':
      action = 'gave a thumbs up';
      icon = '[thumbs up icon]';
      break;
    case 'thumbs-down':
      action = 'gave a thumbs down';
      icon = '[thumbs down icon]';
      break;
    case 'raise-hand':
      action = 'raised their hand';
      icon = '[hand icon]';
      break;
    case 'clap':
      action = 'clapped';
      icon = '[clap icon]';
      break;
    case 'join':
      action = 'joined';
      break;
    case 'leave':
      action = 'left';
      break;
    case 'join-stage':
      action = 'joined the stage';
      break;
    case 'leave-stage':
      action = 'left the stage';
      break;
    default:
      action = "didn't do anything";
      break;
  }

  return (
    <div id={props.id} className="chatBar">
      <span className="personName">{props.name}</span>
      <span className="action">{action}</span>
      <span>{icon}</span>
    </div>
  );
}

export default ChatAction;
