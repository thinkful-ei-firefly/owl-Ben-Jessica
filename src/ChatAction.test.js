import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ChatAction from './ChatAction';

describe('ChatAction component', () => {
  const chatAction = '<ChatAction id=1 name="Test name" />';
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ChatAction />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(chatAction)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });

  it('renders the UI as expected with no unreads', () => {
    const tree = renderer
      .create(chatAction)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
});