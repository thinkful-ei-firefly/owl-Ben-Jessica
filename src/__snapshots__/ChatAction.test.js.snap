// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatAction component renders the UI as expected 1`] = `"<ChatAction id=1 name=\\"Test name\\" />"`;

exports[`ChatAction component renders the UI as expected with no unreads 1`] = `"<ChatAction id=1 name=\\"Test name\\" />"`;
